plugin.json

```diff
@@ -3,6 +3,7 @@
     "name": "Sample Plugin",
     "description": "This plugin serves as a starting point for writing a Mattermost plugin.",
     "version": "0.1.0",
+    "min_server_version": "5.2.0",
```

Comment:
nit: Did 5.2.0 actually support `min_server_version`? Also, if we're depending on 5.12.0 now, should this just get bumped to 5.12.0, and maybe we tag commits supporting an older version?

Reply:
> Did 5.2.0 actually support `min_server_version'.

No. `min_server_version` was added in 5.6.0

> Also, if we're depending on 5.12.0 now, should this just get bumped to 5.12.0, and maybe we tag commits supporting an older version?

I can see four different version we can require here:
1. 5.2.0: This is "real" required server version, as this is required to run the plugin.
2. 5.6.0: Because this added `min_server_version` and the setting will only be taken into account from the version on.
3. 5.10.0: Because this added e.g. asset handling, which we support in the Makefile
4. 5.12.0: Because this is the latest MM version and we "build" on it. And it added support for public files, which we support in the Makefile.

I'm happy to bump this to 5.12 if there a no concerns that this will "limit" plugin development for older servers. Let me know what do you think about this.

Reply:
My gut instinct is to bump this to match `go.mod` and thus `5.12.0`, and iterate separately on how to enable plugin development for older server versions. It might just amount to an FAQ in the README.md describing how to (optionally?) downgrade the `go.mod` dependency and `min_sever_version`, and describing the caveats accordingly.

Reply:
I'm fine with your proposed approach. Would you take care of creating a ticket for the documentation or do the changes yourself?